src/clusterfuzz/_internal/tests/core/bot/fuzzers/centipede/centipede_engine_test.py

```diff
@@ -130,6 +137,20 @@ def _test_reproduce(self,
 
     return re.search(regex, result.output)
 
+  def test_reproduce_uaf_without_unsanitized_binary(self):
```

Comment:
Do you have an idea how long these tests add (especially because they are basically doubled in time because of unshare)?

Reply:
Sure!
I tested it a few times, and the minimum total time cost for these 26 testcases is 36(+-0.5) seconds. This reduces the time consumption from an original value of 202 seconds, with the following changes:
1. [`MAX_TIME`](https://github.com/google/clusterfuzz/blob/35909c46be1d0c6f4e00540dd5f94b80cea2b827/src/clusterfuzz/_internal/tests/core/bot/fuzzers/centipede/centipede_engine_test.py#L56) (to test `fuzz` and `reproduce`) reduced from 25 to 2 seconds.
2. [`_TIMEOUT_PER_INPUT_TEST`](https://github.com/google/clusterfuzz/blob/35909c46be1d0c6f4e00540dd5f94b80cea2b827/src/clusterfuzz/_internal/tests/core/bot/fuzzers/centipede/centipede_engine_test.py#LL64C1-L64C24) reduced from 5 to 2 seconds.
3. [`_CLEAN_EXIT_SECS`](https://github.com/google/clusterfuzz/blob/35909c46be1d0c6f4e00540dd5f94b80cea2b827/src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py#L30) reduced from 10 to 3 seconds.

Using any values lower than those will produce false-positive test failures, because `Centipede` could not launch and produce the error message in time.

Reply:
I am a bit worried about using a low value of [_CLEAN_EXIT_SECS](https://github.com/google/clusterfuzz/blob/35909c46be1d0c6f4e00540dd5f94b80cea2b827/src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py#L30), because it may affect production.

I am a bit inclined to set it to 5 seconds, to give it extra redundancy in production.

Reply:
>202 seconds

I think this is too much time. Thank you for reducing this.

Yeah I would change _CLEAN_EXIT_SECS to 5 seconds. 48 seconds of testing time is an awful lot for a relatively small change. I think in the future we should do more unittesting of centipede changes and less integration testing, since these take very long.

Reply:
> can you use mock to reduce the time just during testing?

I did not think of a way yet, because the slow tests (e.g., timeout) needs to run Centipede and parse its output.